File: src/main/resources/output/IFC4X3_vbnet/Types/IfcSpatialReferenceSelect.vb
' SELECT TYPE IfcSpatialReferenceSelect
' Options:
' - IfcGroup
' - IfcProduct
Public Class IfcSpatialReferenceSelect
    Public ReadOnly Property Value As Object

    Public Sub New(value As Object)
        If Not (TypeOf value Is IfcGroup OrElse TypeOf value Is IfcProduct) Then
            Throw New ArgumentException("Value must be one of: IfcGroup, IfcProduct")
        End If
        Me.Value = value
    End Sub

    Public Function GetValueAs(Of T)() As T
        Return CType(Value, T)
    End Function
End Class

File: src/main/resources/output/IFC4X3_csharp/Types/IfcSpatialReferenceSelect.cs
// SELECT TYPE IfcSpatialReferenceSelect
public class IfcSpatialReferenceSelect {
    public object Value { get; }

    public IfcSpatialReferenceSelect(object value) {
        if (!(value is IfcGroup || value is IfcProduct))
        {
            throw new ArgumentException("Value must be one of: IfcGroup, IfcProduct");
        }
        Value = value;
    }

    public T GetValueAs<T>() {
        return (T)Value;
    }
}
